lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -1267,16 +1267,31 @@ void swift::serialization::diagnoseSerializedASTLoadFailureTransitive(
   }
 }
 
+static std::optional<StringRef> getFlagsFromInterfaceFile(StringRef buffer,
+                                                          StringRef prefix) {
+  StringRef line;
+  unsigned lineCount = 0;
+  const unsigned maxLine = 10;
```

Comment:
It's a bit of a shame that we need a magic constant here, I wonder if there's anything we can do to make sure this is a valid precondition. Maybe asserting on textual interface printing or verification that this always holds true?

Reply:
Let me see what I can do. I am not really eager to land this since I don't see this really impact performance in any significant way in a real world compilation but more performance is always good.

Reply:
Ok, I revisited this patch. Remove the magic number but only parses the first comment block instead.

I don't know what is a good place to implement a diagnostics though. Ideally it should be interface compilation and model comments in swift AST?